scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -255,14 +254,16 @@
 	protected final boolean useTruncatedCertificatePathForVerification;
 
 	private final Set<SessionListener> sessionListeners = new LinkedHashSet<>();
-	
+
 	protected int statesIndex;
 	protected HandshakeState[] states;
 
 	private boolean changeCipherSuiteMessageExpected = false;
 	private boolean sessionEstablished = false;
 	private boolean handshakeAborted = false;
 	private boolean handshakeFailed = false;
+	private boolean pskRequestPending = false;
+	private SecretKey otherSecret;
```

Comment:
should we add some documentation about othersecret ?

> Here "other_secret" either is zeroes (plain PSK case) or comes
   from the Diffie-Hellman or RSA exchange (DHE_PSK and RSA_PSK,
   respectively).  See Sections 3 and 4 for a more detailed
   description.

(source  https://tools.ietf.org/html/rfc4279#section-2 ? )

Reply:
I updated the javadoc with [RFC 5489](https://tools.ietf.org/html/rfc5489#page-4)